Sparc v8 and v9 should still support EM_SPARC binaries, not _just_ SPARC32PLUS.

--- uClibc/ldso/ldso/sparc/dl-sysdep.h	2008-09-15 11:36:11.000000000 -0500
+++ uClibc.bak/ldso/ldso/sparc/dl-sysdep.h	2009-04-08 01:09:53.000000000 -0500
@@ -29,13 +29,8 @@
 /* Here we define the magic numbers that this dynamic loader should accept
  * Note that SPARCV9 doesn't use EM_SPARCV9 since the userland is still 32-bit.
  */
-#if defined(__sparc_v9__) || defined(__sparc_v8__)
 #define MAGIC1 EM_SPARC32PLUS
-#else
-#define MAGIC1 EM_SPARC
-#endif
-
-#undef  MAGIC2
+#define MAGIC2 EM_SPARC
 
 /* Used for error messages */
 #define ELF_TARGET "sparc"
